src/app/private/[uid]/admin/edit-profile/page.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import { prisma } from "@server/db/client";
+import { getServerSession } from "next-auth";
+import { authOptions } from "@api/auth/[...nextauth]/route";
+import EditProfileForm from "./EditProfileForm";
+
+const EditProfile = async () => {
+  let firstName = "";
+  let lastName = "";
+  let pronouns = "";
+  let email = "";
+  let uid = "";
+
+  const fetchProfile = async () => {
```

Comment:
This is along the right approach! We can simplify the code a bit here by interacting with prisma directly on line 8.

```jsx
interface Params {
  params: { // These are the dynamic segments
    uid: string // uid corresponds to [uid] segment
  }
}

const EditProfile = async ({ params }: Params) => {
  const user = await prisma.user.findUniqueOrThrow({ where: { uid: params.uid } });
  ...
}